### PR TITLE
Revert "add back the epsilon value when switching from {oil,gas}-only to both hydrocarbon phases"

### PR DESCRIPTION
**Pull request by andlaus**
this makes flow work for SPE1CASE1 (actually, I do not really know why), but it might lead to different convergence behaviour in some cases.

@totto82: can you test and assess?

### CODE REVIEW

**Comment by atgeirr** [comment]
> this makes flow work for SPE1CASE1 (actually, I do not really know why)

Since this change was made in the original PR to keep behaviour consistent with the existing Flow I am a little skeptical... We must figure out why sufficiently to get both consistent behaviour and a working SPE1CASE1. It's good that you found the source of the trouble!

**Comment by andlaus** [comment]
yes, that's the dilemma I'm worried about, too. Mathematically, not using a "switch hysteresis" should be The Right Thing (TM) to do, right?

**Comment by totto82** [comment]
> yes, that's the dilemma I'm worried about, too. Mathematically, not using a "switch hysteresis" should be The Right Thing (TM) to do, right?

I agree. It is a trick to avoid oscillation between saturated and unsaturated case.

**Comment by totto82** [comment]
Sorry. Hit the wrong button.

**Comment by atgeirr** [comment]
> Mathematically, not using a "switch hysteresis" should be The Right Thing (TM) to do, right?

I guess it could be viewed as a type of damping?

File: ewoms/models/blackoil/blackoilprimaryvariables.hh [hunk 1]
```diff
@@ -461,7 +461,7 @@ public:
                                                                     SoMax);
 
             Scalar Rs = (*this)[Indices::compositionSwitchIdx];
-            if (Rs > (1.0 + eps)*std::min(RsMax, RsSat)) {
+            if (Rs > std::min(RsMax, RsSat)) {
                 // the gas phase appears, i.e., switch the primary variables to { Sw, po,
```
Comment (totto82) [inline ewoms/models/blackoil/blackoilprimaryvariables.hh]: ` Rs > std::min(RsMax, (1.0 + eps)*RsSat)`

Is inconsistent, but works. Shouldn't give any difference for the base case either. 
The same trick can also be applied to Rv.

**Comment by andlaus** [comment]
> I guess it could be viewed as a type of damping?

not really; the update itself isn't altered by it. In this case, it is even less so because in `flow` the epsilon value is only non-zero in the iteration directly after a PV switch IIRC. maybe I got confused by the frequent back-and-forth here, though...

File: ewoms/models/blackoil/blackoilprimaryvariables.hh (hunk 1 above)
Comment (andlaus) [reply ewoms/models/blackoil/blackoilprimaryvariables.hh]: good idea, we have a fallback plan :)

on the other hand, I think that we should try to go the whole way and try to remove the epsilon altogether. Do you remember if doing so leads to any serious regressions or were they just changes that can be attributed to noise?

Comment (totto82) [reply ewoms/models/blackoil/blackoilprimaryvariables.hh]: It gives almost 10% increase on model 2 rel 5. I agree that we should try to remove it, but implementing and testing alternatives is not a trivial task. I suggest merging the fallback plan and think this through later.

Comment (andlaus) [reply ewoms/models/blackoil/blackoilprimaryvariables.hh]: okay, I agree that 10% is too much. Let's go with your suggestion.

**Pull request merged**